source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -743,7 +744,8 @@ void DnsMessageParser::buildResponseBuffer(DnsQueryContextPtr& query_context,
   buffer.writeBEInt<uint16_t>(query_context->response_header_.id);
 
   uint16_t flags;
-  ::memcpy(&flags, static_cast<void*>(&query_context->response_header_.flags), sizeof(uint16_t));
+  ::memcpy(&flags, // NOLINT(safe-memcpy)
```

Comment:
safeMemcpy doesn't work here?

Reply:
On windows, the static_assert fails (as shown below). Should I try with a reinterpret cast of the ```DnsHeaderFlags``` struct or revert it to this change?

From MSVC:
```
static assertion failed, see reference to function template instantiation 'void Envoy::safeMemcpy<uint16_t,Envoy::Extensions::UdpFilters::DnsFilter::DnsHeaderFlags>(T1 *,T2 *)' being compiled
        with
        [
            T1=uint16_t,
            T2=Envoy::Extensions::UdpFilters::DnsFilter::DnsHeaderFlags
        ]

```

Reply:
how about:
```
safeMemcpy(&flags, reinterpret_cast<const uint16_t*>(&query_context->response_header_.flags);
```

Reply:
I am afraid this is breaking. Furthermore, I should check the previous comments since I believe I had tried this before 😅

```
error: taking address of packed member of 'Envoy::Extensions::UdpFilters::DnsFilter::DnsHeader' may result in an unaligned pointer value [-Werror=address-of-packed-member]
  747 |   safeMemcpy(&flags, reinterpret_cast<const uint16_t*>(&(query_context->response_header_.flags)));
      |  
```